### ISSUE TITLE
MIDI-STEP - Xtouch mini - encounting some bugs

### ISSUE THREAD

**Issue by vcvrackideas** [stoermelder/vcvrack-packone#240]
Hi Ben, just bought the best in order to use with midi step and sail as well. The first bug is not even a big deal - after setting relative controls inside mapping software I've got INC and DEC outputs working vice versa, it solves just by switching cables in wrong sail's holes so it's just FYI, little bug.
But the second one is biggiest - midi step doesn't recognize encoder after restarting the Rack anymore. Only after relearning same output and get same CC number it's starting working again. Some kind of initialization but in X-touch mode I guess?
Cheers

**Comment by stoermelder** [comment]
I don't have this controller and can't check myself but MIDI-STEP has not been changed for some time, no one else reported problems. Can you please check the incoming MIDI messages using MIDI-MON?

**Comment by vcvrackideas** [comment]
ok midi-mon see it and decreasing encoder is 127 or 126 and increasing 1 or 2 or 3 sometimes.

**Comment by stoermelder** [comment]
Ok, you can try AKAI MPD218 mode which operates with this values but have you set the controller to "Relative 1"?
![image](https://user-images.githubusercontent.com/47640667/114307810-82efba00-9ae1-11eb-85e3-23b558912a37.png)

**Comment by vcvrackideas** [comment]
Yep I have set it to Relative 1 as in your manual ;-) AkaiMPD works as expected regarding DEC and INC outs but I don't like it due to very small step sizes, x-touch mode serves that well. Again it's not big issue with INC and DEC. The main one is that for example midi-mon responds on the controller moves well from the beginning but midi-step still needs re-learning to "see" X-touch :-(

**Comment by vcvrackideas** [comment]
OK - just made the video regarding this issue - https://youtu.be/rBIueXlO6PY

**Comment by stoermelder** [comment]
Not sure what's going on. I was told that Relative 1 is working the same way as the Arturia Beatstep.  
Can you please create three logs with MIDI-MON using the different relative modes?

**Comment by vcvrackideas** [comment]
Hi Ben sorry for long time no anwer, I created 3 logs and happy to me Relative 2 at least works in right way in terms of INC DEC direction. But the issue with needed relearning encoder each time VCV Rack opens is still persist. Moreover - sometimes when I off VCV Rack window and went to the browsing and then comeback - no reaction from midi controller again, needs to relearn encoder and so on. But MIDI-MON still receives messages anyway anytime so don't know what's goind on with midi step or xtouch.
[xtouch logs.zip](https://github.com/stoermelder/vcvrack-packone/files/6343649/xtouch.logs.zip)

**Comment by stoermelder** [comment]
Ok, thank you for the logs. First of all, it seems I got wrong information in the past about the relative modes of the X-Touch controllers.
But I'm not sure about the relearning of the CCs. After I watched again the video you made I noticed that you are mapping CC1 which is duplicate in slots 1 and 2. This does not work so I fixed that when learning a CC which is already assigned. But I'm not convinced yet this fixed your issue.
